src/Npgsql.NetTopologySuite/NetTopologySuiteHandler.cs

```diff
@@ -0,0 +1,348 @@
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<LineString>.Write(LineString value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<IPolygon>.Write(IPolygon value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<Polygon>.Write(Polygon value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<IMultiPoint>.Write(IMultiPoint value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<MultiPoint>.Write(MultiPoint value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<IMultiLineString>.Write(IMultiLineString value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<MultiLineString>.Write(MultiLineString value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<IMultiPolygon>.Write(IMultiPolygon value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<MultiPolygon>.Write(MultiPolygon value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<IGeometryCollection>.Write(IGeometryCollection value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<GeometryCollection>.Write(GeometryCollection value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task WriteCore(IGeometry value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+        {
+            _writer.Write(value, new WriteStream(buf));
```

Comment:
Probably `WriteStream` should be cached like `LengthStream`.

Reply:
If you refactor it into NpgsqlWriteBuffer (see below) we could cache it there.